Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java

```diff
@@ -32,9 +32,30 @@ public Map<Set<PrimitiveVS>, Guard> getLastUniverseMap() {
         for (Set<PrimitiveVS> set : lastDisjointSet) {
             lastUniverseMap.put(set, universe.get(find(set.iterator().next())));
         }
+        assert(sanityCheck(lastUniverseMap));
         return lastUniverseMap;
     }
 
+    public boolean sanityCheck(Map<Set<PrimitiveVS>, Guard> universeMap) {
```

Comment:
Yo! is this a bug fix?

Reply:
Yes. When dynamic machine creation is involved, [ValueSummaryUnionFind](https://github.com/p-org/P/blob/1fda3058ca3b0bcd28efe494d3d6528bb12a3052/Src/PRuntimes/PSymRuntime/src/main/java/psym/valuesummary/util/ValueSummaryUnionFind.java#L12) breaks and was no longer producing disjoint sets of guards. I corrected this and added this sanity check.